.github/workflows/buildsCI.yaml

```diff
@@ -400,4 +400,46 @@ jobs:
        cmake --build . --config ${{env.LRS_BUILD_CONFIG}} -- -j4
        ls
   
-  
+  cppcheck:
+    name: cppcheck-test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        
+      - name: Check_API
+        shell: bash
+        run: |
+          cd scripts
+          ./api_check.sh
+          cd ..  
+        
+      - name: Prerun
+        shell: bash
```

Comment:
Since Eran is refactoring this file, we may want to consider excluding this sector into it's own yaml,
We can call it SDK-Utilization.
It will start with CppCheck and can later on have address sanitizers and more.
And this is kept for building the SDK on different platforms (`buildsCI`)
What do you think?

Reply:
I think there is logic in splitting to separate Workflows if they don't always run together, like nightly or only for PR's not for every push.
I don't object, but if we will run the static analysis workflow for every push, same as BuildsCI, I don't see the benefit of splitting it. We will have to wait for all tests to finish anyway.

Reply:
Just to make files shorter with context, 1 is building the SDK, 1 is building ROS, 1 is utilization runs.
But if you feel different it's fine by me, your call.

Reply:
Separeted to `static_analysis` workflow.
Don't think `Utilization` is clear enough.